### ISSUE TITLE
Passing settings to `sauceAccount.updateJob()`

### ISSUE THREAD

**Issue by raine** [angular/protractor#326]
Sauce Labs [allows more settings](https://saucelabs.com/docs/additional-config) to annotate jobs than just `passed`.

**Currently:**

```
var cleanUp = function(runner) {
  // ...
  if (sauceAccount) {
    sauceAccount.updateJob(sessionId, {'passed': passed}, function(err) {
```

How do you suggest one would go about implementing this? Something like `sauceAnnotateJob` function in the config that should return an object to be passed to updateJob?

I'm also trying to do this through `jasmineNodeOpts.onComplete` but `sauceAccount` is not visible in that context and it doesn't feel right to make multiple requests.

**Comment by CrispusDH** [comment]
About what are you talking? Sauce Labs just run what you say. If you have some retry mechanism in your test runner it will re run. I'm not sure that understood your issue

**Comment by GuyMograbi** [comment]
Ok. did some investigation. 

If you add `name` to capabilities - it will appear in saucelabs. This works. 

However, protractor's built in support to report failure/success does not work for me.    
Protractor calls `updateJob` but it doesn't really reach the sauce implementation. 

in `runner.js` I added the following log print

```
if (this.driverprovider_.updateJob) {
        console.log('this driverprovider has updateJob, so I am updating', {'passed': testPassed});
         return this.driverprovider_.updateJob({ 'passed': testPassed }).then(() => {
                return this.driverprovider_.teardownEnv();
          });
 }
```

in `dataProviders/dataProvider.js` I added the following print: 

```
 updateJob(update) {
        console.log('nah... not really implemented :D');
        return q.fcall(function () { });
 }
```

I also added a print in `sauce.js` which is never called.    
instead I see 'not really implemented` log print. 

I see saucelab starting a session with the right name, but in the end it has a question mark (which means the status was never reported). 

I am transitioning to using a jasmine reporter.    
Either: https://www.npmjs.com/package/reporter-saucelabs
Or simple implementation of the rest request.
